keepalived/vrrp/vrrp_print.c

```diff
@@ -418,3 +421,182 @@ vrrp_print_stats(void)
 	}
 	fclose(file);
 }
+
+#ifdef _JSON_
+void
+vrrp_print_json(void)
+{
+	FILE *file;
+	file = fopen ("/tmp/keepalived.json","w");
+	if (!file) {
+		log_message(LOG_INFO, "Can't open /tmp/keepalived.json (%d: %s)",
+			errno, strerror(errno));
+		return;
+	}
+
+	list l = vrrp_data->vrrp;
+	if LIST_ISEMPTY(vrrp_data->vrrp) {
+		log_message(LOG_INFO, "vrrp_data->vrrp is empty");
+		return;
+	}
+
+	element e, f;
+	vrrp_t *vrrp;
+	struct json_object *array, *vips;
+	array = json_object_new_array();
+
+	for (e = LIST_HEAD(l); e; ELEMENT_NEXT(e)) {
+		vrrp = ELEMENT_DATA(e);
+		list v = vrrp->vip;
+		struct json_object *instance_json, *json_stats, *json_data;
+		instance_json = json_object_new_object();
+		json_stats = json_object_new_object();
+		json_data = json_object_new_object();
+		vips = json_object_new_array();
+
+		// Dump data to json
+		json_object_object_add(json_data, "iname",
+			json_object_new_string(vrrp->iname));
+		json_object_object_add(json_data, "dont_track_primary",
+			json_object_new_int(vrrp->dont_track_primary));
+		json_object_object_add(json_data, "skip_check_adv_addr",
+			json_object_new_int(vrrp->skip_check_adv_addr));
+		json_object_object_add(json_data, "strict_mode",
+			json_object_new_int(vrrp->strict_mode));
```

Comment:
If compiling with configure option `--enable-conversion-checks` a large number of conversion warnings are generated. This seems to be in the main because the json-c library doesn't have `json_object_new_uint` or `json_object_new_uint64` functions. However, in order to suppress the warnings, the value can either be cast to an int or int64_t, or a unsigned int can be used with the json_object_new_int64() function. I haven't identified all the cases below.